src/main/scala/devices/gpio/IOF.scala

```diff
@@ -63,3 +63,30 @@ object BasePinToIOF {
   }
 }
 
+object InputPortToIOF {
+  def apply(port: Bool, iof: IOFPin): Unit = {
+    val pin = Wire(new BasePin());
+    pin.inputPin(port)
+    BasePinToIOF(pin, iof)
+  }
+
+  def apply(port: UInt, iofs: Vec[IOFPin], offset: Int) {
+    for (i <-0 until port.getWidth) {
```

Comment:
please add one or more require(...) that these ranges are as expected, and that we aren't losing any iofs or will get an array out of bounds error.

Reply:
Added range checking code.